internal/setting/handler.go

```diff
@@ -0,0 +1,312 @@
+package setting
+
+import (
+	"context"
+	"errors"
+	"github.com/NYCU-SDC/clustron-backend/internal/jwt"
+	"github.com/NYCU-SDC/summer/pkg/handler"
+	"github.com/NYCU-SDC/summer/pkg/log"
+	"github.com/NYCU-SDC/summer/pkg/problem"
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgtype"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/ssh"
+	"net/http"
+	"strconv"
+)
+
+type UpdateSettingRequest struct {
+	Username      string `json:"username" validate:"required"`
+	LinuxUsername string `json:"linux_username"`
+}
+
+type SettingResponse struct {
+	Username      string `json:"username"`
+	LinuxUsername string `json:"linuxUsername"`
+}
+
+type AddPublicKeyRequest struct {
+	Title     string `json:"title" validate:"required"`
+	PublicKey string `json:"publicKey" validate:"required"`
+}
+
+type DeletePublicKeyRequest struct {
+	Id string `json:"id" validate:"required,uuid"`
+}
+
+type PublicKeyResponse struct {
+	Id        string `json:"id"`
+	Title     string `json:"title"`
+	PublicKey string `json:"publicLey"`
+}
+
+//go:generate mockery --name Store
+type Store interface {
+	GetSettingByUserId(ctx context.Context, userId uuid.UUID) (Setting, error)
+	UpdateSetting(ctx context.Context, userId uuid.UUID, setting Setting) (Setting, error)
+	GetPublicKeysByUserId(ctx context.Context, userId uuid.UUID) ([]PublicKey, error)
+	GetPublicKeyById(ctx context.Context, id uuid.UUID) (PublicKey, error)
+	AddPublicKey(ctx context.Context, publicKey AddPublicKeyParams) (PublicKey, error)
+	DeletePublicKey(ctx context.Context, id uuid.UUID) error
+}
+
+type Handler struct {
+	validator    *validator.Validate
+	logger       *zap.Logger
+	tracer       trace.Tracer
+	settingStore Store
+}
+
+func validatePublicKey(key string) error {
+	_, _, _, _, err := ssh.ParseAuthorizedKey([]byte(key))
+	if err != nil {
+		return errors.New("invalid public key format")
```

Comment:
We could consider wrapping the original err with `fmt.Errorf("invalid public key format: %w", err)`, so that if something goes wrong, we'll have more detailed error information for easier debugging.